### PR TITLE
Fixes #1736 -  Use constants like MyBB::INPUT_INT instead of the raw val...

### LINKED ISSUE #1736: Use constants like MyBB::INPUT_INT instead of the raw value

### PR DESCRIPTION
**Pull request by ATofighi**
...ue #1736 
https://gist.github.com/ATofighi/1e77e8001682e32891ae

### CODE REVIEW

**Comment by Destroy666x** [comment]
I don't think it's still complete. At least your script didn't complete the `(int)$mybb->input` conversion, check for example admin/modules/user/groups.php

**Comment by ATofighi** [comment]
@Destroy666x Fixed.
now I only don't change `(int)$mybb->input[....][....]`s.

File: modcp.php [hunk 1]
```diff
@@ -1362,9 +1362,9 @@
 			'uid' => $mybb->user['uid'],
 			'subject' => $mybb->input['title'],
 			'message' => $mybb->input['message'],
-			'allowhtml' => (int)$mybb->input['allowhtml'],
-			'allowmycode' => (int)$mybb->input['allowmycode'],
-			'allowsmilies' => (int)$mybb->input['allowsmilies'],
```
Comment (ATofighi) [inline modcp.php]: I think it should be bool . :-\
@Destroy666x @mybb/developers

Comment (Destroy666x) [inline modcp.php]: I think it's ok as it is now.

File: admin/modules/config/profile_fields.php [hunk 2]
```diff
@@ -212,7 +212,7 @@
 	$selected_values = '';
 	if($mybb->input['viewableby'] != '' && $mybb->input['viewableby'] != -1)
 	{
-		$selected_values = explode(',', (string)$mybb->input['viewableby']);
+		$selected_values = explode(',', $mybb->get_input('viewableby', MyBB::INPUT_STRING));
```
Comment (Sama34) [inline admin/modules/config/profile_fields.php]: There is no need for `MyBB::INPUT_STRING`.

Comment (ATofighi) [inline admin/modules/config/profile_fields.php]: you're right.
fixed.

**Pull request merged**